lib/iex/lib/iex/pry.ex

```diff
@@ -22,7 +22,14 @@ defmodule IEx.Pry do
   def pry(binding, %Macro.Env{} = env) do
     %{file: file, line: line, module: module, function: function_arity} = env
     self = self()
-    opts = [binding: binding, dot_iex_path: "", env: env, prefix: "pry"]
+
+    stacktrace =
+      case Process.info(self, :current_stacktrace) do
+        {:current_stacktrack, stacktrace} -> prune_stacktrace(stacktrace)
+        _ -> nil
```

Comment:
I don't think this clause can ever be reached, so we can remove it and write it as:

    {:current_stacktrack, stacktrace} = Process.info(self, :current_stacktrace)
    [..., stacktrace: prune_stacktrace(stacktrace)]